### DISCUSSION TITLE
trying to understand some of the uniforms in old-film filter

### DISCUSSION THREAD

**Discussion by mreinstein** [pixijs/pixijs#9206] [Q&A] [answered]
I'm porting a couple pixi filters from glsl to webgpu/wgsl, and I'm confused on some of these uniforms.

The current shader I'm looking at is the old film grain.

What is `filterArea` in this case?  I've looked at the code a little and my impression was the filterArea specified the rectangle of the screen affected by the current shader, but that doesn't seem to be the case here https://github.com/pixijs/filters/blob/main/filters/old-film/src/old-film.frag#L94

`seed` is clearly some kind of random number related field, I'm assuming based on the name it's only set once, or is this updated every frame? Is this intended to be a value from `0` -> `1` ?

**Comment by SuperSodaSea** [comment, marked as answer]
`filterArea` is `vec4(dstWidth, dstHeight, srcX, srcY)`, so `filterArea.xy` is the size of the screen:

https://github.com/pixijs/pixijs/blob/313789dbe3bd19506a3a21ffe59aa4731f462288/packages/core/src/filters/FilterSystem.ts#L299-L302

For the uniform `seed`, if you don't change its value, you can get the same noise for each frame. If you want the noise to stay the same, just leave it alone. If you want different noise every frame, you can update it with random number.

**Reply by mreinstein** [reply]
I know  there isn't a right or wrong answer in terms of how one packs a 4 element array with values, but I was expecting `filterArea[0]` to be `sourceFrame.x`, `filterArea[1]` to be `sourceFrame.y`.  

The consequence of this layout is in the shader `filterArea.xy` ends up being `width, height` rather than `x,y`.  

I'm not pretending to be the 1st person to encounter this, but coming in fresh, this definitely tripped me up.

Thanks for the clarification!
